compiler/src/iree/compiler/Codegen/Common/test/convert_to_destination_passing_style.mlir

```diff
@@ -162,6 +162,71 @@ func.func @matmul_inplace() {
 
 // -----
 
+#pipeline_layout = #hal.pipeline.layout<
+    bindings = [
+        #hal.pipeline.binding<storage_buffer, "ReadOnly|Indirect">,
+        #hal.pipeline.binding<storage_buffer, "ReadOnly|Indirect">,
+        #hal.pipeline.binding<storage_buffer, Indirect>],
+    flags = Indirect>
+func.func @matmul_forall() {
+  %c4 = arith.constant 4 : index
+  %c32 = arith.constant 32 : index
+  %c0_i32 = arith.constant 0 : i32
+  %c0 = arith.constant 0 : index
+  %0 = hal.interface.binding.subspan layout(#pipeline_layout) binding(0) alignment(64) offset(%c0) flags("ReadOnly|Indirect") : !flow.dispatch.tensor<readonly:tensor<62x32x16x32xi8>>
+  %1 = hal.interface.binding.subspan layout(#pipeline_layout) binding(1) alignment(64) offset(%c0) flags("ReadOnly|Indirect") : !flow.dispatch.tensor<readonly:tensor<126x32x16x32xi8>>
+  %2 = hal.interface.binding.subspan layout(#pipeline_layout) binding(2) alignment(64) offset(%c0) flags(Indirect) : !flow.dispatch.tensor<readwrite:tensor<62x126x16x16xi32>>
+  %3 = flow.dispatch.tensor.load %0, offsets = [0, 0, 0, 0], sizes = [62, 32, 16, 32], strides = [1, 1, 1, 1] : !flow.dispatch.tensor<readonly:tensor<62x32x16x32xi8>> -> tensor<62x32x16x32xi8>
+  %4 = flow.dispatch.tensor.load %1, offsets = [0, 0, 0, 0], sizes = [126, 32, 16, 32], strides = [1, 1, 1, 1] : !flow.dispatch.tensor<readonly:tensor<126x32x16x32xi8>> -> tensor<126x32x16x32xi8>
+  %5 = tensor.empty() : tensor<62x126x16x16xi32>
+  %6 = scf.forall (%arg0, %arg1) = (0, 0) to (62, 126) step (2, 2) shared_outs(%arg2 = %5) -> (tensor<62x126x16x16xi32>) {
+    %extracted_slice_1 = tensor.extract_slice %arg2[%arg0, %arg1, 0, 0] [2, 2, 16, 16] [1, 1, 1, 1] : tensor<62x126x16x16xi32> to tensor<2x2x16x16xi32>
+    %11 = linalg.fill ins(%c0_i32 : i32) outs(%extracted_slice_1 : tensor<2x2x16x16xi32>) -> tensor<2x2x16x16xi32>
+    %12 = scf.for %arg3 = %c0 to %c32 step %c4 iter_args(%arg4 = %11) -> (tensor<2x2x16x16xi32>) {
+      %extracted_slice_2 = tensor.extract_slice %3[%arg0, %arg3, 0, 0] [2, 4, 16, 32] [1, 1, 1, 1] : tensor<62x32x16x32xi8> to tensor<2x4x16x32xi8>
+      %13 = tensor.empty() : tensor<2x4x16x32xi8>
+      %14 = linalg.copy ins(%extracted_slice_2 : tensor<2x4x16x32xi8>) outs(%13 : tensor<2x4x16x32xi8>) -> tensor<2x4x16x32xi8>
+      %extracted_slice_3 = tensor.extract_slice %4[%arg1, %arg3, 0, 0] [2, 4, 16, 32] [1, 1, 1, 1] : tensor<126x32x16x32xi8> to tensor<2x4x16x32xi8>
+      %15 = linalg.copy ins(%extracted_slice_3 : tensor<2x4x16x32xi8>) outs(%13 : tensor<2x4x16x32xi8>) -> tensor<2x4x16x32xi8>
```

Comment:
Can you simplify this test case a bit. A single input binding and a single copy should be good enough.

Reply:
+1, we should keep the test as simple as possible for maintenance.